k9mail-library/src/main/java/com/fsck/k9/mail/ssl/DefaultTrustedSocketFactory.java

```diff
@@ -160,7 +169,18 @@ public Socket createSocket(Socket socket, String host, int port, String clientCe
         SSLSocketFactory socketFactory = sslContext.getSocketFactory();
         Socket trustedSocket;
         if (socket == null) {
-            trustedSocket = socketFactory.createSocket();
+            if (proxySettings.enabled) {
+                InetSocketAddress proxyAddress = new InetSocketAddress(proxySettings.host, proxySettings.port);
+                Proxy proxy = new Proxy(Type.SOCKS, proxyAddress);
+
+                Socket underlying = new Socket(proxy);
+                InetSocketAddress serverAddress = new InetSocketAddress(host, port);
+                underlying.connect(serverAddress, SOCKET_CONNECT_TIMEOUT);
+
+                trustedSocket = socketFactory.createSocket(underlying, proxySettings.host, proxySettings.port, true);
+            } else {
```

Comment:
The question of self-signed certificates for proxy implementations comes to mind. Should we be using our own socket factory here.